### ISSUE TITLE
Touch events?

### ISSUE THREAD

**Issue by kccqzy** [nteract/semiotic#293]
I noticed that this library has good support for mouse events: the automatic hover tooltips and customizable behavior on hover. Any plans to add touch events? At a minimum, maybe just allowing the user to specify `customTouchBehavior` that is called when a touch starts/ends? I imagine this could be very similar, except it just listens on `onTouchStart`, `onTouchMove` and `onTouchEnd`.

**Comment by emeeks** [comment]
This is a good point and I'll make sure to handle these.

**Comment by kccqzy** [comment]
While implementing this myself, I encountered this react issue: https://github.com/facebook/react/issues/6436

The gist is that since iOS 11.3 the touch event handlers are by default passive, so things like preventDefault won't work. This can cause some unwanted scrolling. 

Might be relevant once you get to implement this issue. Anyways, thanks for the great library!

**Comment by kccqzy** [comment]
Okay so we have implemented a very terrible hack. It's just a MVP to get hover annotations (tooltip) working: 

1. In `onTouchStart`, we call `this.changeVoronoi(overlayRegion.onMouseEnter(), props.hoverAnnotation)`.
2. In `onTouchMove`, we do some hacky detection of whether or not the finger has been moved to a different overlay region. Basically, even if the finger has moved, the `onTouchMove` events still carry with them their original target. We then compare the real target to obtain the real overlay region (call it `realOverlayRegion`), and then call `this.changeVoronoi(realOverlayRegion.onMouseEnter(), props.hoverAnnotation)`.
3. In `onTouchEnd`, we call `this.changeVoronoi()`.

It's some pretty terrible code, but at least it's working for us
